### PR TITLE
Remove unused token_type_ids in MPNet

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

This PR adds a warning when the argument `token_type_ids` is given a show a message saying that this argument is never used. I just supressed the internal appearance of this argument without modifying the method signatures in order to do not integrates a breaking change.

Should I update the Tokenizer to make it returns only `attention_mask`?

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
cc @LysandreJik But I don't think this is the way to go. We should just silently remove the `token_type_ids` from the tokenizer and the model args at the same time. I don't think this warrants a deprecation cycle since it's not used.

**Comment by jplu** [comment]
I'm totally fine to definitely suppress this argument in once if this is prefered.

File: src/transformers/models/mpnet/modeling_tf_mpnet.py [hunk 1]
```diff
@@ -582,6 +578,7 @@ def call(
             training=training,
             kwargs_call=kwargs,
         )
+        print(inputs)
```
Comment (LysandreJik) [inline src/transformers/models/mpnet/modeling_tf_mpnet.py:581]: No need for this!

**Review by LysandreJik — COMMENTED** [review]
I don't think we need to put warnings however, as @sgugger said I would simply remove these from both the tokenizer and the forward call. Their presence is a bug, and not a feature we're modifying.

**Review by sgugger — COMMENTED** [review]
Thanks for adapting @jplu!

Not an expert on the tokenization part, but is the method `build_inputs_with_special_tokens` still necessary in that case? (It's in both tokenizers files.)

**Comment by jplu** [comment]
Not an expert neither. Maybe @LysandreJik knows better.

**Comment by patrickvonplaten** [comment]
> Thanks for adapting @jplu!
> 
> Not an expert on the tokenization part, but is the method `build_inputs_with_special_tokens` still necessary in that case? (It's in both tokenizers files.)

I think it's still required as it puts *e.g.* the [sep] token correctly between two sentences. I don't think that `build_inputs_with_special_tokens` necessarily has something to do with `input_type_ids`

**Review by patrickvonplaten — APPROVED** [review]

File: src/transformers/models/mpnet/modeling_tf_mpnet.py [hunk 2]
```diff
@@ -156,7 +155,7 @@ def call(
         Get token embeddings of inputs
 
         Args:
-            inputs: list of three int64 tensors with shape [batch_size, length]: (input_ids, position_ids, token_type_ids)
+            inputs: list of three int64 tensors with shape [batch_size, length]: (input_ids, position_ids)
```
Comment (LysandreJik) [inline src/transformers/models/mpnet/modeling_tf_mpnet.py:158]:
```suggestion
            inputs: list of two int64 tensors with shape [batch_size, length]: (input_ids, position_ids)
```

**Review by LysandreJik — APPROVED** [review]
Great, LGTM!

**Pull request merged**